### ISSUE TITLE
Assembly load in StampSvn fails since Fody 1.26.2.0

### ISSUE THREAD

**Issue by krk** [Fody/Fody#153]
```
2>MSBUILD : warning : Fody/StampSvn: GetSvnInfo error: System.IO.FileLoadException: Could not load file or assembly 'SharpSvn, Version=1.7013.2566.15257, Culture=neutral, PublicKeyToken=d729672594885a28' or one of its dependencies. Attempt to load an unverifiable executable with fixups (IAT with more than 2 sections or a TLS section.) (Exception from HRESULT: 0x80131019)
2>MSBUILD : warning : File name: 'SharpSvn, Version=1.7013.2566.15257, Culture=neutral, PublicKeyToken=d729672594885a28' ---> System.IO.FileLoadException: Could not load file or assembly 'SharpSvn, Version=1.7013.2566.15257, Culture=neutral, PublicKeyToken=d729672594885a28' or one of its dependencies. Attempt to load an unverifiable executable with fixups (IAT with more than 2 sections or a TLS section.) (Exception from HRESULT: 0x80131019)
2>MSBUILD : warning : File name: 'SharpSvn, Version=1.7013.2566.15257, Culture=neutral, PublicKeyToken=d729672594885a28' ---> System.IO.FileLoadException: Attempt to load an unverifiable executable with fixups (IAT with more than 2 sections or a TLS section.) (Exception from HRESULT: 0x80131019)
2>MSBUILD : warning :    at System.Reflection.RuntimeAssembly.nLoadImage(Byte[] rawAssembly, Byte[] rawSymbolStore, Evidence evidence, StackCrawlMark& stackMark, Boolean fIntrospection, SecurityContextSource securityContextSource)
2>MSBUILD : warning :    at System.Reflection.Assembly.Load(Byte[] rawAssembly)
2>MSBUILD : warning :    at InnerWeaver.LoadFromFile(String assemblyPath) in c:\TeamCity\buildAgent\work\ed448661dbb30d2e\FodyIsolated\AssemblyLoader.cs:line 35
2>MSBUILD : warning :    at InnerWeaver.CurrentDomain_AssemblyResolve(Object sender, ResolveEventArgs args) in c:\TeamCity\buildAgent\work\ed448661dbb30d2e\FodyIsolated\InnerWeaver.cs:line 31
2>MSBUILD : warning :    at System.AppDomain.OnAssemblyResolveEvent(RuntimeAssembly assembly, String assemblyFullName)
2>MSBUILD : warning : 
2>MSBUILD : warning : 
2>MSBUILD : warning :    at SvnHelper17.GetSvnInfo(String targetFolder)
2>MSBUILD : warning :    at SvnHelper.GetSvnInfo(String targetFolder)
2>MSBUILD : warning :    at ModuleWeaver.Execute()
2>MSBUILD : warning : 
2>MSBUILD : warning : 
2>      svnInfo found.
2>MSBUILD : error : Fody: An unhandled exception occurred:
2>MSBUILD : error : Exception:
2>MSBUILD : error : Object reference not set to an instance of an object.
2>MSBUILD : error : StackTrace:
2>MSBUILD : error :    at ModuleWeaver.Execute()
2>MSBUILD : error :    at lambda_method(Closure , Object )
2>MSBUILD : error :    at InnerWeaver.ExecuteWeavers(List`1 weaverInstances) in c:\TeamCity\buildAgent\work\ed448661dbb30d2e\FodyIsolated\InnerWeaver.cs:line 111
2>MSBUILD : error :    at InnerWeaver.Execute() in c:\TeamCity\buildAgent\work\ed448661dbb30d2e\FodyIsolated\InnerWeaver.cs:line 51
2>MSBUILD : error : Source:
2>MSBUILD : error : StampSvn.Fody
2>MSBUILD : error : TargetSite:
2>MSBUILD : error : Void Execute()
2>MSBUILD : error :

**Comment by SimonCropp** [comment]
can u put a repro in svn somewhere?

**Comment by krk** [comment]
I have uploaded a repro, you can get it [here](http://s000.tinyupload.com/?file_id=58268974778208163709).

**Comment by krk** [comment]
I can repro it with Fody 1.27.0 as well.

**Comment by krk** [comment]
Can repro it with Fody 1.28.0 as well.

**Comment by SimonCropp** [comment]
the stack trace i get is 
```
Error	2	Fody: An unhandled exception occurred:
Exception:
Object reference not set to an instance of an object.
StackTrace:
   at ModuleWeaver.Execute() in c:\Downloads\Stamp-master\Fody\ModuleWeaver.cs:line 99
   at lambda_method(Closure , Object )
   at InnerWeaver.ExecuteWeavers() in c:\ConsoleBuildAgent\work\ed448661dbb30d2e\FodyIsolated\InnerWeaver.cs:line 135
   at InnerWeaver.Execute() in c:\ConsoleBuildAgent\work\ed448661dbb30d2e\FodyIsolated\InnerWeaver.cs:line 53
Source:
StampSvn.Fody
TargetSite:
Void Execute()
	FodyAssemblyLoad_repro_153
```
So looking at your code you have 
```
        VersionInfo ver = null;
        try
        {
            ver = SvnHelper.GetSvnInfo(svnDir);
        }
        catch (Exception ex)
        {
            LogWarning("GetSvnInfo error: " + ex.ToString());
        }

        LogInfo("svnInfo found.");

        assemblyVersion = ModuleDefinition.Assembly.Name.Version;
```
Note that you swallow the excepion and then use the variable. So if an exception does occue you will get a nullreference exception. And sure enough if I look in the build output I can see

```
1>MSBUILD : warning : Fody/StampSvn: GetSvnInfo error: System.IO.FileLoadException: Could not load file or assembly 'SharpSvn, Version=1.7013.2566.15257, Culture=neutral, PublicKeyToken=d729672594885a28' or one of its dependencies. Attempt to load an unverifiable executable with fixups (IAT with more than 2 sections or a TLS section.) (Exception from HRESULT: 0x80131019)
1>MSBUILD : warning : File name: 'SharpSvn, Version=1.7013.2566.15257, Culture=neutral, PublicKeyToken=d729672594885a28' ---> System.IO.FileLoadException: Could not load file or assembly 'SharpSvn, Version=1.7013.2566.15257, Culture=neutral, PublicKeyToken=d729672594885a28' or one of its dependencies. Attempt to load an unverifiable executable with fixups (IAT with more than 2 sections or a TLS section.) (Exception from HRESULT: 0x80131019)
1>MSBUILD : warning : File name: 'SharpSvn, Version=1.7013.2566.15257, Culture=neutral, PublicKeyToken=d729672594885a28' ---> System.IO.FileLoadException: Attempt to load an unverifiable executable with fixups (IAT with more than 2 sections or a TLS section.) (Exception from HRESULT: 0x80131019)
1>MSBUILD : warning :    at System.Reflection.RuntimeAssembly.nLoadImage(Byte[] rawAssembly, Byte[] rawSymbolStore, Evidence evidence, StackCrawlMark& stackMark, Boolean fIntrospection, SecurityContextSource securityContextSource)
1>MSBUILD : warning :    at System.Reflection.Assembly.Load(Byte[] rawAssembly)
1>MSBUILD : warning :    at InnerWeaver.LoadFromFile(String assemblyPath) in c:\ConsoleBuildAgent\work\ed448661dbb30d2e\FodyIsolated\AssemblyLoader.cs:line 35
1>MSBUILD : warning :    at InnerWeaver.CurrentDomain_AssemblyResolve(Object sender, ResolveEventArgs args) in c:\ConsoleBuildAgent\work\ed448661dbb30d2e\FodyIsolated\InnerWeaver.cs:line 34
1>MSBUILD : warning :    at System.AppDomain.OnAssemblyResolveEvent(RuntimeAssembly assembly, String assemblyFullName)
```

**Comment by SimonCropp** [comment]
so i think this https://github.com/Fody/Fody/commit/67df1523adf7631dd93aff0ab0f1fd7d5a54eb9a is conflicting with your assembly resolve. i will have a go at fixing it on my side

**Comment by kalisohn** [comment]
i still have the same issue with Fody 1.29.4

**Comment by r0f3n** [comment]
After a day hunting this one down I'll post my fix in case someone else stumbles across this post with the same problem.
I had the problem with VS 2015 build tools. Main thing I needed to do was upgrade to Build Tools Update 3. After that I found another problem with the local repository being in an old format, so I needed to delete and get the repo again. (Jenkins has a default setting with version 1.4).
